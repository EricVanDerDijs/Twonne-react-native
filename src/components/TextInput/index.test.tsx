import React from 'react';
import { TextInput } from './index';
import renderer from 'react-test-renderer';

let textValue = '';

const mockHandler = (text: string): void  => {
  // tslint:disable-next-line
  textValue = text;
};

it('Renders our custom TesxtInput', () => {
  const tree = renderer.create((
    <TextInput
      value=''
      onChangeText={mockHandler}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Updates TesxtInput value correctly', () => {
  const tree = renderer.create((
    <TextInput
      value=''
      onChangeText={mockHandler}
    />
  ));

  const textInputInstance = tree.getInstance();

  if (textInputInstance
  && textInputInstance.props
  && typeof textInputInstance.props.onChangeText === 'function') {
    textInputInstance.props.onChangeText('test text');
  }

  expect(textValue).toBe('test text');
});
