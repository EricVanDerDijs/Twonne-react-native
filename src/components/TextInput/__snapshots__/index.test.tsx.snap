// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders our custom TesxtInput 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[Function]}
  rejectResponderTermination={true}
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 30,
    }
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;
